nav2_behavior_tree/src/behavior_tree_engine.cpp

```diff
@@ -55,7 +55,12 @@ BehaviorTreeEngine::run(
 
       onLoop();
 
-      loopRate.sleep();
+      if (!loopRate.sleep()) {
+        RCLCPP_WARN(
+          rclcpp::get_logger("BehaviorTreeEngine"),
+          "Behavior Tree tick rate %0.2f was exceeded!",
+          1.0 / (loopRate.period().count() * 1.0e-9));
+      }
```

Comment:
It means that we didn't sleep for the full period that we're eating into the next iteration of the loop. If you want to submit a PR to adjust to `... tick rate <> was not reached.` I'd merge that in

Reply:
@maksymdidukh can you take that please?

Reply:
> @maksymdidukh would you like to take that?

sure!

Reply:
Btw, same as [here](https://github.com/ros-planning/navigation2/pull/3076), it would be useful to know what the **achieved frequency** is to know how bad it is. I didn't manage to do it reliably in the un-merged PR and gave up, maybe you manage :wink: